src/jdk.compiler/share/classes/com/sun/tools/javac/comp/LambdaToMethod.java

```diff
@@ -755,14 +755,6 @@ private void addDeserializationCase(MethodHandleSymbol refSym, Type targetType,
         String functionalInterfaceClass = classSig(targetType);
         String functionalInterfaceMethodName = samSym.getSimpleName().toString();
         String functionalInterfaceMethodSignature = typeSig(types.erasure(samSym.type));
-        Symbol baseMethod = refSym.baseSymbol();
-        Symbol origMethod = baseMethod.baseSymbol();
-        if (baseMethod != origMethod && origMethod.owner == syms.objectType.tsym) {
-            //the implementation method is a java.lang.Object method transferred to an
-            //interface that does not declare it. Runtime will refer to this method as to
-            //a java.lang.Object method, so do the same:
-            refSym = ((MethodSymbol) origMethod).asHandle();
-        }
```

Comment:
javac workaround for existing behavior, as implemented by JDK-8282080. No longer useful (and, in fact, actively bad, because it's anticipating the wrong result from `getDeclaringMethod`).